cmd/kubeadm/app/cmd/init.go

```diff
@@ -354,6 +355,19 @@ func (i *Init) Run(out io.Writer) error {
 	if err := controlplanephase.CreateInitStaticPodManifestFiles(manifestDir, i.cfg); err != nil {
 		return fmt.Errorf("error creating init static pod manifest files: %v", err)
 	}
+
+	etcdDataDir := i.cfg.Etcd.Local.DataDir
+
+	if err := os.MkdirAll(etcdDataDir, 0700); err != nil {
+		return fmt.Errorf("failed to create directory %q: %v", etcdDataDir, err)
+	}
+
+	// Set SELinux context for etcd data directory
+	selinuxContext := kubeadmsystem.NewSELinuxContext()
+	if err := kubeadmsystem.SetSELinuxFilecon(etcdDataDir, selinuxContext); err != nil {
+		return fmt.Errorf("error setting SELinux context for etcd data directory: %v", err)
+	}
+
```

Comment:
This whole block needs to be moved inside the block of `if i.cfg.Etcd.External == nil {`. We need this in the case of local etcd.
In fact, it can crash at the moment, due to `i.cfg.Etcd.Local == nil` (the external etcd case).